rust/hook-worker/src/worker.rs

```diff
@@ -173,39 +189,160 @@ impl<'p> WebhookWorker<'p> {
+    let mut kafka_ack_futures = Vec::new();
+    for (result, mut metadata) in iter::zip(results, metadata_vec) {
+        match result {
+            Ok(result) => {
+                if hog_mode {
+                    if let Some(payload) = create_hoghook_kafka_payload(result, &mut metadata).await
+                    {
+                        match kafka_producer.send_result(FutureRecord {
+                            topic: cdp_function_callbacks_topic,
+                            payload: Some(&payload),
+                            partition: None,
+                            key: None::<&str>,
+                            timestamp: None,
+                            headers: None,
+                        }) {
+                            Ok(future) => kafka_ack_futures.push(future),
+                            Err((error, _)) => {
+                                error!("error producing kafka message: {}", error);
+                                sleep(Duration::from_secs(30)).await;
+                                return; // Return early to avoid committing the batch.
+                            }
+                        };
+                    }
                 }
+            }
+            Err(e) => {
+                error!("error processing webhook job: {}", e)
+            }
+        }
+    }
+
+    for result in join_all(kafka_ack_futures).await {
+        match result {
+            Ok(Ok(_)) => {}
+            Ok(Err((error, _))) => {
+                error!("error awaiting kafka ack: {}", error);
+                sleep(Duration::from_secs(30)).await;
+                return; // Return early to avoid committing the batch.
+            }
+            Err(_) => {
+                // Cancelled due to timeout while retrying
+                error!("error producing kafka message: timed out");
+                sleep(Duration::from_secs(30)).await;
+                return; // Return early to avoid committing the batch.
+            }
+        }
+    }
 
-                let results = join_all(futures).await;
-                for result in results {
-                    if let Err(e) = result {
-                        error!("error processing webhook job: {}", e);
+    let _ = batch.commit().await.map_err(|e| {
+        error!("error committing transactional batch: {}", e);
+    });
+}
+
+async fn create_hoghook_kafka_payload(
+    result: WebhookResult,
+    metadata: &mut Value,
+) -> Option<String> {
+    if let Value::Object(ref mut object) = metadata {
+        // Add the response or error in the `asyncFunctionResponse` field.
+        match result {
+            WebhookResult::Success(response) => {
+                let async_function_response = json!({
+                    "timings": [{
+                        "kind": "async_function",
+                        "duration_ms": response.duration.as_millis().try_into().unwrap_or(u32::MAX)
+                    }],
+                    "response": {
```

Comment:
@benjackwhite You mentioned maybe renaming `vmResponse` to `response` so I'm using that here. Easy to change if you've changed your mind.

Reply:
No thats perfect (I just need to do a PR to change it)